src/Yoti/Util/Age/AbstractAgeProcessor.php

```diff
@@ -2,40 +2,85 @@
 
 namespace Yoti\Util\Age;
 
+use Yoti\Entity\Attribute;
+use Yoti\Entity\AgeVerification;
+
 abstract class AbstractAgeProcessor
 {
-    // You must define this pattern in the child class
-    const AGE_PATTERN = '';
+    // You could re-define this value in the child class
+    const AGE_DELIMITER = ':';
 
-    public $profileData;
+    /**
+     * @var Attribute
+     */
+    protected $attribute;
 
-    public function __construct(array $profileData)
+    /**
+     * AbstractAgeProcessor constructor.
+     *
+     * @param Attribute $attribute
+     */
+    public function __construct(Attribute $attribute)
     {
-        $this->profileData = $profileData;
+        $this->attribute = $attribute;
     }
 
     /**
-     * Get age attribute and value.
+     * Return true if attribute name matches the pattern.
      *
-     * @return null|array
+     * @param Attribute $attribute
+     *
+     * @return bool
      */
-    public function getAgeRow()
+    protected function isDerivedAttribute(Attribute $attribute)
```

Comment:
Would be clearer if it was called _handlesAttribute_ or _processesAttribute_

Reply:
I would prefer to keep it as it is, as this is returning a boolean so that name reflects what it's doing.